### PR TITLE
Add Contributor Documentation

### PR DESCRIPTION
**Pull request by katbyte**

### CODE REVIEW

**Review by stephybun — APPROVED** [review]
LGTM 🍑

File: contributing/topics/guide-new-fields-to-data-source.md [hunk 1]
```diff
@@ -1,3 +1,80 @@
...
+				check.That(data.ResourceName).Key("new_property").HasValue("1"),
+			),
+		},
+	})
+}
+```
+
+## Docs
+
+* Lastly, don't forget to update the docs where the property ordering is determined alpahbetically.
```
Comment (tombuildsstuff) [inline contributing/topics/guide-new-fields-to-data-source.md:77]:
```suggestion
* Lastly, don't forget to update the docs where the property ordering is determined alphabetically.
```

File: contributing/topics/guide-typed-model.md [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: guide for adopting model objects linked to from the Data Source/Resource code
```
Comment (mbfrahry) [inline contributing/topics/guide-typed-model.md:1]: Is this something we want to keep in here for ourselves to know what to add later?

File: contributing/topics/maintainer-changelog.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Maintainer Specific: Updating the Changelog
```
Comment (mbfrahry) [inline contributing/topics/maintainer-changelog.md:1]: We call out `Maintainer Specific` here but I'd even add an extra note to say this is not for people looking to contribute only but instead for the maintainers of the provider instead. I just want to be extraaaaa clear that normal contributors do not need to read/do this

File: contributing/topics/our-recommendations-for-opening-a-pr.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Recommendations for sending a Pull Request
```
Comment (mbfrahry) [inline contributing/topics/our-recommendations-for-opening-a-pr.md:1]: Another TODO doc

File: contributing/topics/reference-acceptance-testing.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+The bare-minimum example for this is provisioning the `basic` configuration and then updating it using the `complete` test configuration above, for example:
+
+```go
+func TestAccExampleResource_update(t *testing.T) {
+        data := acceptance.BuildTestData(t, "azurerm_example_resource", "test")
+        r := ExampleResourceTest{}
+
+        data.ResourceTest(t, r, []acceptance.TestStep{
+            {
+				// first provision the resource
```
Comment (mbfrahry) [inline contributing/topics/reference-acceptance-testing.md:291]: Spacing

File: contributing/topics/reference-acceptance-testing.md [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+                Config: r.basic(data),
+                Check: acceptance.ComposeTestCheckFunc(
+                    check.That(data.ResourceName).ExistsInAzure(r),
+                ),
+            },
+            data.ImportStep(),
+            {
+				// then perform the update
```
Comment (mbfrahry) [inline contributing/topics/reference-acceptance-testing.md:299]: Spacing

**Review by mbfrahry — APPROVED** [review]
LGTM just lots of weird spacing issues and maybe some leftover todos that need another gandre

**Review by mbfrahry — APPROVED** [review]
LGTM with a todo that needs looked at

**Pull request merged**